modules/dockermodelrunner/dockermodelrunner.go

```diff
@@ -0,0 +1,134 @@
+// Container represents the DockerModelRunner container type used in the module
+type Container struct {
+	*socat.Container
+	model   string
+	baseURL string
+}
+
+// Run creates an instance of the DockerModelRunner container type.
+func Run(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*Container, error) {
+	return RunWithImage(ctx, socat.DefaultImage, opts...)
+}
+
+// RunWithImage creates an instance of the DockerModelRunner container type, using a specific image for the socat container.
+func RunWithImage(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*Container, error) {
+	settings := defaultOptions()
+
+	// Process model runner options.
+	for _, opt := range opts {
+		if apply, ok := opt.(Option); ok {
+			if err := apply(&settings); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// Add socat options, which are applied to the socat container.
+	opts = append(opts, testcontainers.WithWaitStrategy(
+		wait.ForListeningPort("80/tcp"),
+		wait.ForHTTP("/").WithPort("80/tcp").WithStatusCodeMatcher(func(status int) bool {
+			return status == http.StatusOK
+		}).WithResponseMatcher(func(body io.Reader) bool {
+			bodyBytes, err := io.ReadAll(body)
+			if err != nil {
+				return false
+			}
+			return strings.Contains(string(bodyBytes), "The service is running")
+		}),
+	))
+	opts = append(opts, socat.WithTarget(socat.NewTarget(modelRunnerPort, modelRunnerEntrypoint)))
+
+	socatCtr, err := socat.Run(ctx, img, opts...)
+	var c *Container
+	if socatCtr != nil {
+		c = &Container{Container: socatCtr, model: settings.model}
+	}
+
+	if err != nil {
+		return c, fmt.Errorf("socat run: %w", err)
+	}
+
+	c.baseURL = socatCtr.TargetURL(modelRunnerPort).String()
+
+	if settings.model != "" {
+		err := c.PullModel(ctx, settings.model)
+		if err != nil {
+			return c, fmt.Errorf("pull model: %w", err)
+		}
+	}
+
+	return c, nil
+}
+
+// InspectModel returns a model that is already pulled using the Docker Model Runner format.
+func (c *Container) InspectModel(ctx context.Context, namespace string, name string) (types.ModelResponse, error) {
+	dmrClient := client.NewClient(c.baseURL)
+
+	modelResponse, err := dmrClient.InspectModel(ctx, namespace, name)
+	if err != nil {
+		return modelResponse, fmt.Errorf("list models: %w", err)
+	}
+
+	return modelResponse, nil
+}
+
+// ListModels lists all models that are already pulled using the Docker Model Runner format.
+func (c *Container) ListModels(ctx context.Context) ([]types.ModelResponse, error) {
+	dmrClient := client.NewClient(c.baseURL)
+
+	var models []types.ModelResponse
+	var err error
+
+	models, err = dmrClient.ListModels(ctx)
+	if err != nil {
+		return models, fmt.Errorf("list models: %w", err)
+	}
+
+	return models, nil
+}
+
+// PullModel pulls a model from the Docker Model Runner
+func (c *Container) PullModel(ctx context.Context, model string) error {
+	log.Default().Printf("🙏 Pulling model %s. Please be patient, no progress bar yet!", model)
```

Comment:
suggestion: Definitely agree a progress bar option would be good the amount of time needed

Reply:
@kiview I followed your example. Is it it, or the backend not providing the progress bar?